model/metrics/clr-metrics.yaml

```diff
@@ -0,0 +1,167 @@
+groups:
+  - id: metric.clr.gc.collections.count
+    type: metric
+    metric_name: clr.gc.collections.count
+    brief: "Number of garbage collections that have occurred since the process has started."
+    instrument: counter
+    unit: "{collection}"
+    stability: experimental
+    attributes:
+      - ref: clr.gc.generation
+        requirement_level: required
+
+  - id: metric.clr.gc.objects.size
+    type: metric
+    metric_name: clr.gc.objects.size
+    brief: "The number of bytes currently allocated on the managed GC heap. Fragmentation and other GC committed memory pools are excluded."
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.allocations.size
+    type: metric
+    metric_name: clr.gc.allocations.size
+    brief: >
+      The number of bytes allocated on the managed GC heap since the process has
+      started. The returned value does not include any native allocations.
+    instrument: counter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.committed_memory.size
+    type: metric
+    metric_name: clr.gc.committed_memory.size
+    brief: >
+      The amount of committed virtual memory for the managed GC heap, as
+      observed during the latest garbage collection.
+    note: >
+      Committed virtual memory may be larger than the heap size because it
+      includes both memory for storing existing objects (the heap size) and some
+      extra memory that is ready to handle newly allocated objects in the
+      future.
+    instrument: updowncounter
+    unit: "By"
+    stability: experimental
+
+  - id: metric.clr.gc.heap.size
```

Comment:
I don't know how JVM memory model works, `gc.heap` does sound more accurate to me than `dotnet.heap` since there are heaps managed by .NET GC and heaps not managed by the .NET GC.

Reply:
there is a balance between accuracy and user-friendliness.

E.g. if I'm looking at all metrics, how can I find the memory utilization. I'd be surprised to find it in `anything.gc.memory`. 
Most of the users (myself included) rarely care about subtle technical differences. 

If we had a metric that showed memory  usage that users can slice and dice based on additional dimensions, it could be both, technically precise and also friendly.

In the case of Java metrics, they kinda achieved it in [`jvm.memory.used`](https://github.com/open-telemetry/semantic-conventions/blob/main/docs/runtime/jvm-metrics.md#metric-jvmmemoryused):
- you can sum up by memory type to get both heap and non-heap
- you can get heap only and break it down by GC generation
- ...

If multiple metrics are provided, we'd depend on the user to know all the different memory consumption sources - as a typical non-advanced user, I won't be able to answer a simple question of how much memory my .NET app consumes and how the break down by different memory types looks like.

Reply:
I believe this is discussed in https://github.com/open-telemetry/semantic-conventions/pull/1035/files#r1649489585, resolving this discussion to consolidate things